Import required packages

In [19]:
import numpy as np
import pandas as pd

Collect the feature names and the output name

In [28]:
features = []
output = ''

with open('adult.names') as infile:
    info = infile.readlines()
    output = info[94][:-2]
    for line in info[96:]:
        words = line.split()
        features.append(words[0])
        
        
for i in range(len(features)):
    features[i] = features[i][:-1]
        
print("Feature names:\n", features, '\n')
print("Output name:\n", output)

Feature names:
 ['age', 'workclass', 'fnlwgt', 'education', 'education-num', 'marital-status', 'occupation', 'relationship', 'race', 'sex', 'capital-gain', 'capital-loss', 'hours-per-week', 'native-country'] 

Output name:
 >50K, <=50K


Read the data from file

In [29]:
df = pd.read_csv('adult.data')
df.columns = features + [output]

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32560 entries, 0 to 32559
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32560 non-null  int64 
 1   workclass       32560 non-null  object
 2   fnlwgt          32560 non-null  int64 
 3   education       32560 non-null  object
 4   education-num   32560 non-null  int64 
 5   marital-status  32560 non-null  object
 6   occupation      32560 non-null  object
 7   relationship    32560 non-null  object
 8   race            32560 non-null  object
 9   sex             32560 non-null  object
 10  capital-gain    32560 non-null  int64 
 11  capital-loss    32560 non-null  int64 
 12  hours-per-week  32560 non-null  int64 
 13  native-country  32560 non-null  object
 14  >50K, <=50K     32560 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


| Continuous (N = 32560) |  Mean (SD) |  Median (IQR) |  Min to Max |
|---|---|---|---|
|  Age (years) | 30.6 (13.6)  | 37 (28-48)  |  17-90 |
| Education number (years)  | 10.1 (2.6)  |  10 (9-12)  | 1-16  |
|  Capital gain | 1077.6 (7385.4)  |  0 (0-0) | 0-99999  |
|  Capital loss | 87.3 (403.0) | 0 (0-0)  |  0-4356 |
| Hours per week   | 40.4 (12.3)  | 40 (40-45)  |  1-99 |

<hr style="border:2px solid gray"> </hr>

| Categorical | N/32560 (%)  |
|---|---|
|  **Work** |   |
| Private  |  |
|  Self-employed, not inc. |   |
|  Self-employed, inc. |   |
| Federal government  |   |
| Local government  |   |
|  State government |   |
|  Without pay |   |
| Never worked  |   |
| | |
|  **Education** |   |
|Bachelors | |
|Some college | |
|11th grade | |
|High school | |
| Professional school| |
|Assoc-acdm | |
|Assoc-voc | |
|9th grade | |
|7th-8th grade | |
|12th grade | |
|Masters | |
| 1st-4th grade| |
| 10th grade| |
|Doctorate | |
|5th-6th grade | |
|Preschool | |
| | |
|**Marital status** | |
|Married-civ-spouse | |
|Divorced | |
|Never-married | |
| Separated| |
|Widowed | |
|Married-spouse-absent | |
|Married-AF-spouse | |
| | |
| **Relationship**| |
|Wife | |
| Own-child| |
|Husband | |
|Not in family | |
| Other relative| |
| Unmarried| |
| | |
|**Occupation** | |
|Tech-support | |
| Craft-repair| |
|Other-service | |
| Sales| |
| Exec-managerial| |
|Prof-specialty | |
|Handlers | |
|Machine-op-inspct | |
|Adm-clerical | |
|Farming-fishing | |
|Transport-moving | |
| Priv-house-serv| |
|Protective-serv | |
| Armed-Forces| |
| | |
|**Race** | |
|White | |
| Asian Pacific islander| |
| American Indian eskimo| |
| Other| |
|Black | |
| | |
| **Sex**| |
| Female| |
| Male| |
| | |
| **Native country**| |
| United-States| |
|Cambodia | |
|England | |
|Puerto Rico | |
|Canada | |
| Germany| |
|Outlying-US (Guam, USVI, etc.) | |
| India| |
| Japan| |
|Greece | |
| South| |
| China| |
| Cuba| |
|Iran| |
| Honduras| |
| Philippines| |
|Italy | |
|Poland | |
|Jamaica | |
| Vietnam| |
| Mexico| |
|Portugal | |
|Ireland | |
| France| |
|Dominican Republic | |
|Laos | |
|Ecuador | |
|Taiwan | |
|Haiti | |
| Columbia| |
|Hungary | |
|Guatemala | |
|Nicaragua | |
|Scotland | |
| Thailand| |
|Yugoslavia | |
|El-Salvador | |
|Trinadad & Tobago | |
|Peru | |
|Hong | |
| Netherlands| |
| | |
|**Income**< | |
|\>50k | |
|<=50k | |
| | |

